library/dicom_frame.php

```diff
@@ -17,10 +17,10 @@
 
 require_once('../interface/globals.php');
 
-$web_path = $_REQUEST['web_path'];
 $patid = $_REQUEST['patient_id'];
 $docid = isset($_REQUEST['document_id']) ? $_REQUEST['document_id'] : $_REQUEST['doc_id'];
-$web_path .= '&retrieve&patient_id=' . attr($patid) . '&document_id=' . attr($docid) . '&as_file=false'
+$d = new Document(attr($docid));
+$web_path = attr(str_ireplace($server_document_root, '', $d->get_url_filepath()));
```

Comment:
Hi @sjpadgett ,
Is this just doing a direct link to a document within the documents directory? If so, then this won't work on a properly configured OpenEMR instance(if this did work, then would mean all the patient documents would be open to the public without any authentication protection). Also note get_url_filepath() is not the mechanism to find the path of a document and additionally does not cover couchdb; will be much easier if get the&retrieve... link to support this (see retrieve_action function in controllers/C_Document.class.php script for details)

Reply:
Guess i'm missing something with this class and will look at code again. All i'm interested in here is the file link from table.  Also, for sure this will break file security and I put this up to demo that it works and how I had to handle file urls to get it to work. I've noted this on Victors PR and hoped to point out that the viewer runs completely on client side requiring url file access to the dicom files. I haven't found a way around this. Just to be clear, the viewer needs to inspect the file url to determine what loader(zip,raw etc) to use to load (xhr) the file into working arrays thus, file retrieval via the controller breaks this requirement.